proposals/0836-lean-core-jsc.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Extract JSC (JavaScriptCore) from React Native Core
+author:
+  - Kudo Chien <kudo@expo.dev>
+date: 2024-11-28
+---
+
+# RFC0836: Extract JSC (JavaScriptCore) from React Native Core
+
+## Summary
+
+This RFC proposes a Lean Core effort to extract JSC (JavaScriptCore) from React Native's core.
+
+## Proposed Changes and Timeline
+
+- Version 0.78: Remove the [`jsc-android` npm dependency](https://github.com/facebook/react-native/blob/2d337efc23c662143b3f39a6c994d80fec047054/packages/react-native/package.json#L132) from React Native and instead download the artifact from Maven Central. When using Hermes, downloading the unused npm dependency is unnecessary and removing it saves about 32 MB of uncompressed build time size.
+- Version 0.80: Introduce a new `react-native-javascriptcore` npm package with a newer version of JavaScriptCore for Android. In this version, we will only provide the [Intl variant](https://github.com/react-native-community/jsc-android-buildscripts/blob/9c61fece4753902a2cd6d29dfa46b7b521f0c821/README.md#international-variant), as newer JavaScriptCore versions do not support disabling Intl. The existing JSCRuntime from the core should still function with this version but will display a deprecation warning during build time. A challenge will be managing the core JSCRuntime alongside the third-party JSCRuntime from `react-native-javascriptcore`; details will be mentioned in the [Mixing `JSCRuntime` from core and third-party](#mixing-jscruntime-from-core-and-third-party) section.
+  - Could we support `react-native-javascriptcore` only on New Architecture mode?
```

Comment:
in 0.80 I would also love to effectively stop testing JSC in release testing:
https://github.com/reactwg/react-native-releases/blob/main/docs/guide-release-testing.md#dimensions-to-test

Could you add it here as things to do/consider?

Reply:
Also yes, we can restrict `react-native-javascriptcore` to be New Architecture only